plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require("fs");
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+  constructor(extensionContext) {
+    if (extensionContext !== undefined) {
+      this._extensionContext = extensionContext;
+      this._tmpLogLines = [];
+      this._currentIndex = null;
+      this._fileIndex = 0;
+      this._writeToFileTimer = null;
+      this._timerActivated = false;
+      fs.mkdir(this._extensionContext.logUri.fsPath, (err) => {
+        if (!err) {
+          this.writeHeaderLine();
+        }
+      });
+      this._performanceTxtPath = path.join(
+        this._extensionContext.logUri.fsPath,
+        "qljsPerformanceTrace.txt"
+      );
+
+      this._writableStream = fs.createWriteStream(this._performanceTxtPath, {
+        encoding: "utf8",
+        flags: "a",
+      });
+      this._currentSessionMap = {};
+    }
+  }
+
+  fileNameOptimizer(fileName) {
+    if (
+      Object.prototype.hasOwnProperty.call(this._currentSessionMap, fileName)
+    ) {
+      this._currentIndex = this._currentSessionMap[fileName];
+    } else {
+      this._fileIndex++;
+      this._currentSessionMap[fileName] = this._fileIndex;
+      this._currentIndex = this._fileIndex;
+      this.writeNewFileLine(fileName);
+    }
+  }
+
+  logLintDuration(timeTaken, fileName) {
+    this.fileNameOptimizer(fileName);
+    this.writeExistingFileLine(timeTaken);
+    this.setWriteToFileTimer();
+  }
+
+  writeHeaderLine() {
+    this._tmpLogLines.push(
+      "timestamp",
+      "file-ref",
+      "event-id",
+      "file-name-or-duration"
+    );
+  }
+
+  writeNewFileLine(fileName) {
+    this._tmpLogLines.push(Date.now(), this._currentIndex, "n", fileName);
+  }
+
+  writeExistingFileLine(timeTaken) {
+    this._tmpLogLines.push(Date.now(), this._currentIndex, "p", timeTaken);
+  }
+
+  setWriteToFileTimer() {
+    if (!this._timerActivated) {
+      setTimeout(() => {
+        this._timerActivated = true;
+        this.writeToFile();
+      }, 1000);
+    }
+  }
+
+  writeToFile() {
+    while (this._tmpLogLines.length > 0) {
+      let tracingLine = this._tmpLogLines.splice(0, 4);
+      if (typeof tracingLine[0] === "string") {
```

Comment:
I think a variable or comment would significantly help with readability:
```suggestion
      let isHeaderLine = typeof tracingLine[0] === "string";
      if (isHeaderLine) {
```